commands/operator-sdk/cmd/scorecard/basic_tests.go

```diff
@@ -19,179 +19,82 @@ import (
-		}
-		if !mapType {
-			if err := runtimeClient.Update(context.TODO(), obj); err != nil {
-				return fmt.Errorf("failed to update object: %v", err)
-			}
-			err = wait.Poll(time.Second*1, time.Second*15, func() (done bool, err error) {
-				err = runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}, obj)
-				if err != nil {
-					return false, err
-				}
-				return !reflect.DeepEqual(statCopy, obj.Object["status"]), nil
-			})
-		}
-		if err != nil {
-			return err
-		}
-		//reset stat copy to match
-		statCopy = make(map[string]interface{})
-		for k, v := range obj.Object["status"].(map[string]interface{}) {
-			statCopy[k] = v
-		}
+	if res.EarnedPoints != 1 {
+		res.Suggestions = append(res.Suggestions, "Add a 'status' field to your Custom Resource")
 	}
-	return nil
+	return res
 }
 
-// wiritingIntoCRsHasEffect simply looks at the proxy logs and verifies that the operator is sending PUT
+// writingIntoCRsHasEffect simply looks at the proxy logs and verifies that the operator is sending PUT
 // and/or POST requests to the API server, which should mean that it is creating or modifying resources.
-func writingIntoCRsHasEffect(obj *unstructured.Unstructured) (string, error) {
-	test := scorecardTest{testType: basicOperator, name: "Writing into CRs has an effect", maximumPoints: 1}
+func (t *WritingIntoCRsHasEffectTest) Run(ctx context.Context) *TestResult {
+	res := &TestResult{Test: t, MaximumPoints: 1}
 	kubeclient, err := kubernetes.NewForConfig(kubeconfig)
 	if err != nil {
-		return "", fmt.Errorf("failed to create kubeclient: %v", err)
+		res.Errors = append(res.Errors, fmt.Errorf("failed to create kubeclient: %v", err))
+		return res
 	}
 	dep := &appsv1.Deployment{}
-	err = runtimeClient.Get(context.TODO(), types.NamespacedName{Namespace: obj.GetNamespace(), Name: deploymentName}, dep)
+	err = t.Client.Get(ctx, types.NamespacedName{Namespace: t.CR.GetNamespace(), Name: deploymentName}, dep)
 	if err != nil {
-		return "", fmt.Errorf("failed to get newly created operator deployment: %v", err)
+		res.Errors = append(res.Errors, fmt.Errorf("failed to get newly created operator deployment: %v", err))
+		return res
 	}
 	set := labels.Set(dep.Spec.Selector.MatchLabels)
 	pods := &v1.PodList{}
-	err = runtimeClient.List(context.TODO(), &client.ListOptions{LabelSelector: set.AsSelector()}, pods)
+	err = t.Client.List(ctx, &client.ListOptions{LabelSelector: set.AsSelector()}, pods)
 	if err != nil {
-		return "", fmt.Errorf("failed to get list of pods in deployment: %v", err)
+		res.Errors = append(res.Errors, fmt.Errorf("failed to get list of pods in deployment: %v", err))
+		return res
 	}
 	proxyPod = &pods.Items[0]
-	req := kubeclient.CoreV1().Pods(obj.GetNamespace()).GetLogs(proxyPod.GetName(), &v1.PodLogOptions{Container: "scorecard-proxy"})
+	// this is a temporary workaround; will be fixed with PR #1027
+	t.ProxyPod = proxyPod
```

Comment:
Is `t.ProxyPod` used anywhere outside the scope of this `Run()` function? If not, is there a reason it is a field on the test struct or can it just be a local variable?

Reply:
The updated scorecard resources check (PR #1027) moves the usage of the proxy pod to a new function so it can be used by multiple tests. When that PR is merged, we will only be setting `t.ProxyPod` here:  https://github.com/operator-framework/operator-sdk/pull/1027/files#diff-26de46567cad99950f4cceb66603d685R155

Then we will use it for multiple tests that will use the proxy's logs.